src/extension.ts

```diff
@@ -110,26 +112,43 @@ function initCommand(context: vscode.ExtensionContext, commandId: string, callba
 function initAsyncCommand(context: vscode.ExtensionContext, commandId: string, callback: (...args: any[]) => Promise<any>) {
 	context.subscriptions.push(vscode.commands.registerCommand(commandId, async (...args: any[]) => {
 		const start = Date.now();
-		let result = 'Succeeded';
-		let errorData: string = '';
+		let properties: { [key: string]: string; } = {};
+		properties.result = 'Succeeded';
+		let errorData: ErrorData | undefined = null;
+		const output = util.getOutputChannel();
+		context.subscriptions.push(output);
 
 		try {
 			await callback(...args);
 		} catch (err) {
-			result = 'Failed';
-			errorData = util.errToString(err);
+			properties.result = 'Failed';
 			if (err instanceof UserCancelledError) {
-				result = 'Canceled';
+				properties.result = 'Canceled';
 			}
 			else if (err instanceof Error) {
```

Comment:
This should be an 'else' case. All errors or strings or whatever should go through ErrorData